### DISCUSSION TITLE
Too many changes when scalafmt is upgraded

### DISCUSSION THREAD

**Discussion by yadavan88** [scala-steward-org/scala-steward#3085] [Q&A] [answered]
Hi,
I noticed a weird problem multiple times when there is a scala-steward pr for scalafmt update. 
Even in minor version update, sometimes, there are a huge number of formatting changes. However, if I run scalafmt before and after the update, there are no formatting diff. This diff is there only in the scala-steward PR.
Here is a sample for reference: https://github.com/Baeldung/scala-tutorials/pull/730

Could someone help me to understand what is going on here? I wonder how there could be so many diff when I cant find anything if I run scalafmt manually. 

Thanks in advance !

**Comment by fthomas** [comment, marked as answer]
I suspect that the difference comes from how you run scalafmt and how Scala Steward runs it. Scala Steward uses the scalafmt binary (probably installed via Coursier) and runs that to reformat **all** files in the project. @yadavan88, you probably use scalafmt via the sbt plugin and run `sbt scalafmt` to reformat only files **that are part of the sbt build**.

**Reply by yadavan88** [reply]
Oh... Yeah, that makes sense. I didn't think of `sbt scalafmt` vs `scalafmt` difference. Thanks for that info :) 

Do you know if it is possible somehow to configure that only the sbt managed files to be formatted? If not, does it make sense to have this feature? 
For now, I will run the native app for scalafmt to format non-sbt managed files as well, so that I wont run into this problem in future. 

@fthomas  Really appreciate the answer :)

**Reply by fthomas** [reply]
> Do you know if it is possible somehow to configure that only the sbt managed files to be formatted?

Yes, you could modify `.scalafmt.conf` to exclude directories that are not part of the sbt build.
